src/dag/dag.cpp

```diff
@@ -320,6 +323,8 @@ std::shared_ptr<DagManager> DagManager::getShared() {
 void DagManager::stop() {
   unique_lock lock(mutex_);
   trx_mgr_ = nullptr;
+  stopped_ = true;
+  block_worker_.join();
```

Comment:
I think this should be: `if (block_worker_.joinable()) {  block_worker_.join(); }` + you should call stop() function in destructor as you are handling new thread and without explicitly calling `stop()` function (lets say in test) there would be a crash that might be hard to debug.